src/utils/wallet-seed.js

```diff
@@ -109,5 +112,12 @@ async function deriveEncryptionKey(password, salt, iterations, digest) {
 }
 
 export function lockWallet() {
-  setUnlockedMnemonicAndSeed(null, null);
+  setUnlockedMnemonicAndSeed(null, null, null);
+}
+
+// Returns the 32 byte key used to encrypt imported private keys.
+async function deriveImportsPrivateKey(mnemonic) {
```

Comment:
Can this do the derivation from the seed instead of the mnemonic? Then it doesn't need to be async.